File: Stack/Implementation.js
class Stack {
  constructor(size) {
    this.size = size;
    this.top = -1;
    this.arr = new Array(size);
  }
  push(element) {
    if (this.top < this.size - 1) {
        this.arr[++this.top] = element;
    //   this.top++;
    //   this.arr[this.top] = element;
    } else {
      console.log("Stack Overflow");
      return;
    }
  }
  pop() {
    if (this.top < 0) {
      console.log("Stack Underflow");
      return;
    } else {
      this.top--;
      return this.arr[this.top];
    }
  }
  peek(){
    if (this.top >=0){
        console.log("Top element is: " + this.arr[this.top]);
        return this.arr[this.top];
    }else{
       return -1;
    }
  }
  isEmpty(){
    if(this.top<0){
        console.log("Stack is empty");
    }
    else{
        console.log("Stack is not empty");
    }
  }
  displayData() {
    for (let i = this.top; i >= 0; i--) {
        console.log(this.arr[i] + " ");
    }
    // console.log();
}
}
let st = new Stack(4);
st.push(1);
st.push(2);
st.push(3);
st.push(4);
// st.push(5);
st.displayData();
st.pop();
st.pop();
st.pop();
st.pop();
st.pop();
console.log(st.peek());
st.isEmpty();
st.displayData();

// let st = new Stack(4);
// st.push(1);
// st.push(2);
// st.push(3);
// st.push(4);
// st.push(5);
// console.log(st.pop());
// console.log(st.pop());
// console.log(st.peek());
// console.log(st.isEmpty());

File: Stack/implementation.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 10

typedef struct
{
    int items[MAX];
    int top;
} Stack;
void create_stack(Stack *stack)
{
    stack->top = -1;
}
int isFull(Stack *stack)
{
   return stack->top == MAX - 1;
}
void push_stack(Stack *stack, int item)
{
    if (isFull(stack))
    {
        printf("Stack is full! Can't push  elements %d\n", item);
    }
    else
    {
        stack->items[++stack->top] = item; // Corrected the indexing
        printf("%d pushed onto stack\n", item);
    }
}

int pop_stack(Stack *stack)
{
    if (stack->top == -1)
    {
        printf("Stack is empty\n");
        return -1;
    }
    int item = stack->items[stack->top];
    stack->top--;
    return item;
}
int peek_stack(Stack *stack)
{
    if (stack->top == -1)
    {
        printf("Stack is empty\n");
        return -1;
    }
    return stack->items[stack->top];
}
void display_stack(Stack *stack)
{
    if (stack->top == -1)
    {
        printf("Stack is empty\n");
    }
    for (int i = stack->top; i >= 0; i--)
    {
        printf("%d ", stack->items[i]);
    }
    printf("\n");
}
int main()
{
    Stack stack;
    create_stack(&stack);

    push_stack(&stack, 10);
    push_stack(&stack, 20);
    push_stack(&stack, 30);
    push_stack(&stack, 40);
    push_stack(&stack, 50);
    push_stack(&stack, 60);
    push_stack(&stack, 70);
    push_stack(&stack, 80);
    push_stack(&stack, 90);
    push_stack(&stack, 100);
    push_stack(&stack, 110);

    printf("Top element is %d\n", peek_stack(&stack));

    printf("Stack before popping elements:\n");
    display_stack(&stack);

    printf("%d popped from stack\n", pop_stack(&stack));
    printf("%d popped from stack\n", pop_stack(&stack));

    printf("Stack after popping elements:\n");
    display_stack(&stack);

    return 0;
}
